experimental/BasisLieHighestWeight/src/WeylPolytope.jl

```diff
@@ -53,38 +53,108 @@ function convert_lattice_points_to_monomials(
 end
 
 @doc raw"""
-    get_lattice_points_of_weightspace(weights::Vector{Vector{QQFieldElem}}, wght::Vector{QQFieldElem}) -> Vector{Vector{ZZRingElem}}
+    get_lattice_points_of_weightspace(weight_roots::Vector{Vector{QQFieldElem}}, L::LieAlgebraStructure, weight::Vector{ZZRingElem}, highest_weight::Vector{ZZRingElem},
+
 
 Calculates all lattice points in a given weightspace for a Lie algebra highest weight module.
 This is equivalent to finding $\mathbb{Z}$-linear combinations of `weights` that equal `wght`.
 All weights are given as coefficients to the simple roots $\alpha_i$.
 """
 function get_lattice_points_of_weightspace(
-  weights::Vector{Vector{QQFieldElem}}, wght::Vector{QQFieldElem}
+  weight_roots::Vector{Vector{QQFieldElem}}, 
+  L::LieAlgebraStructure, 
+  weight::Vector{ZZRingElem}, 
+  highest_weight::Vector{ZZRingElem}, 
+  zero_coordinates::Set{Any}
 )
   # calculate all integer solutions to the following linear program:
   # [   |              |    ]       [   x   ]      
-  # [weights[1]...weights[k]]   *   [   |   ]   =   weight 
+  # [weight_roots[1]...weight_roots[k]]   *   [   |   ]   =   weight
   # [   |              |    ]       [  res  ] 
   # [   |              |    ]       [   |   ]
   # where res[i] >= 0 for all i
-
-  n = length(weights)
+  wght = w_to_alpha(L, weight )
+  n = length(weight_roots)
   m = length(wght)
-  A = zero_matrix(QQ, 2m + n, n)
-  b = [zero(QQ) for _ in 1:(2m + n)]
+ 
+  #zero_coordinates = 0
+  #zero_coordinates = compute_zero_coordinates(weight_roots, L, weight, highest_weight) 
+  A = zero_matrix(QQ, 2m + n + length(zero_coordinates), n)
+  b = [zero(QQ) for _ in 1:(2m + n + length(zero_coordinates))]
+
   # equalities
   for i in 1:n
-    w = matrix(QQ, m, 1, weights[i])
+    w = matrix(QQ, m, 1, weight_roots[i])
     A[1:m, i] = w
     A[(m + 1):(2m), i] = -w
   end
+
   b[1:m] = wght
   b[(m + 1):(2m)] = -wght
   # non-negativity
   for i in 1:n
     A[2m + i, i] = -1
   end
-
-  return Vector{ZZRingElem}.(lattice_points(polyhedron(A, b)))
+  j = 0
+  for i in zero_coordinates 
+    j = j+1
+    A[2m + n + j, i] = 1
+    b[2m + n + j] = 0
+  end
+  sol = Vector{ZZRingElem}.(lattice_points(polyhedron(A, b)))
+  return sol
 end
+
+@doc raw"""
+compute_zero_coordinates(
+  weight_roots::Vector{Vector{QQFieldElem}}, L::LieAlgebraStructure, weight::Vector{ZZRingElem}, highest_weight::Vector{ZZRingElem},
+)
```

Comment:
```suggestion
    compute_zero_coordinates(weight_roots::Vector{Vector{QQFieldElem}}, L::LieAlgebraStructure, weight::Vector{ZZRingElem}, highest_weight::Vector{ZZRingElem})

```
im docstring muss die signatur einzeilig sein und mit genau 4 Spaces eingerückt. Anschließend eine Leerzeile